### ISSUE TITLE
LibreView import's time formats need a wider support

### ISSUE THREAD

**Issue by mrinnetmaki** [tidepool-org/uploader#1461]
Commit https://github.com/tidepool-org/uploader/pull/1431/commits/309b5753b99b8d64df84b73e5f66061cb9fd7a82 lists two available time formats. However, in LibreView you can set your preference in even more formats.

![image](https://user-images.githubusercontent.com/5329276/147107921-4c4b71cf-5a83-4d53-a536-d1f9e11c1601.png)

Not sure whether sundial can tell the difference between Month-Day-Year and Day-Month-Year from the data file, or whether we should read the file and 

Having downloaded data in each formats, I can see (in different files) at least:

`12-20-2021 23:15`
`12-20-2021 11:15 PM`
`20-12-2021 23:15`
`20-12-2021 11:15 PM`
`2021-12-20 23:15`

I guess the right time formats should rather be something like

```
const LIBREVIEW_TS_FORMAT = [
    'MM/DD/YYYY hh:mm A',
    'DD/MM/YYYY HH:mm',
    'YYYY-MM-DD HH:mm',
    'YYYY-MM-DD HH:mm A',
    'DD-MM-YYYY HH:mm',
    'DD-MM-YYYY HH:mm A',
    'MM-DD-YYYY HH:mm',
    'MM-DD-YYYY HH:mm A',
  ];
```

(Not sure which formats are available in US instance, for instance.)

As said, however, I'm unsure whether sundial would manage to tell the difference between DD-MM-YYYY and MM-DD-YYYY.

**Comment by mrinnetmaki** [comment]
Ping @gniezen and @kpouget.

**Comment by gniezen** [comment]
Ah, this would explain #1459.

Hmm, we may need to use the locale to distinguish between DD-MM-YYYY and MM-DD-YYYY then. :grimacing:

**Comment by mrinnetmaki** [comment]
Or peek to the file and see which kind of entries it contains?

**Comment by gniezen** [comment]
@mrinnetmaki I think the only solution here that doesn't involve guessing (and getting it wrong if there are not enough entries) is to enforce the YYYY-MM-DD date format.

**Comment by GermoVeltmaat** [comment]
Assuming that there are more days in the file, you can iterate over the entries en check the possible day and month entries. If one is more then 12 then that is the day.
If all entries are 12 or less then the import could generate an error
(just my contribution)

**Comment by gniezen** [comment]
This issue will be addressed by https://github.com/tidepool-org/uploader/pull/1472.

**Comment by gniezen** [comment]
Should be fixed in Uploader v2.44.0 that was just released.
